### ISSUE TITLE
Still a bug in issue #166

### ISSUE THREAD

**Issue by bellezaodiada** [CindyJS/CindyJS#635]
Problem with triangles and lines outstanding from issue #166
Sadly it seems that although the problem with lines being cut is fixed, the coloured polygons still do not fill with colour in the js version

**Comment by bellezaodiada** [comment]
Here is the cdy file. The planes of the two desargue triangles are coloured correctly when Java is used, but not when saved as the javascript version. Also the text is not correctly shown in UTF5, the accented characters come out as question marks.

[desargues3d.zip](https://github.com/CindyJS/CindyJS/files/1496829/desargues3d.zip)

**Comment by kranich** [comment]
@bellezaodiada I think you have closed this accidently.

**Comment by bellezaodiada** [comment]
Yes, that was a mistake. I did not mean to close the issue. Many thanks.

**Comment by kranich** [comment]
@bellezaodiada Have you used the latest Cinderella version, 2.9 build 1901, that can be downloaded from https://beta.cinderella.de/?

As far as I can see, everything works as expected.

Cinderella 2.9, build 1901:
![Cinderella 2.9, build 1901](https://user-images.githubusercontent.com/7249754/33141199-b1ec1cb6-cfb2-11e7-8142-8636d38aa320.png)


Exported and opened in Firefox 57.0:
![Exported and opened in Firefox 57.0](https://user-images.githubusercontent.com/7249754/33141101-6159314e-cfb2-11e7-90a4-a45ea8ded344.png)

And opened in Chromium Version 62.0.3202.94:
![opened in Chromium Version 62.0.3202.94](https://user-images.githubusercontent.com/7249754/33141313-13fc151e-cfb3-11e7-8fd1-7f367617280d.png)

**Comment by kranich** [comment]
Ah, now I see, the segment between G and P is not visible!

**Comment by kranich** [comment]
Is that what you meant?

**Comment by kranich** [comment]
```

**Comment by kranich** [comment]
@strobelm Maybe you want to have a look at this since you have already worked on the related issue #166?

**Comment by bellezaodiada** [comment]
I was using build 1835. I have now loaded 1901 and it seems to work perfectly, including the segment GP, on Firefox 57.0 (64 bit) and Google Chrome Version 62.0.3202.94 (Official Build) (64-bit) and Microsoft Edge 41.16299.15.0

**Comment by bellezaodiada** [comment]
Firefox 57.0 (64 bit)
![image](https://user-images.githubusercontent.com/16744124/33141992-3b6ef54c-cfb5-11e7-8048-882155967836.png)

**Comment by bellezaodiada** [comment]
Fantastic news. Many thanks to everyone involved.

**Comment by bellezaodiada** [comment]
However, there are still a few unexplained problems. For example the following diagram looks like this in Cinderella:
![image](https://user-images.githubusercontent.com/16744124/33152261-1cf5de88-cfdc-11e7-96ca-bf0fcdaf9762.png)

But when exported to CindyJS appears as 
![image](https://user-images.githubusercontent.com/16744124/33152402-addcd76c-cfdc-11e7-9308-caf988005642.png)

Am I doing something wrong here?

**Comment by bellezaodiada** [comment]
The source file is 
[pappusparab.zip](https://github.com/CindyJS/CindyJS/files/1497622/pappusparab.zip)

The same problem seems to be happening with
[pappushyper.zip](https://github.com/CindyJS/CindyJS/files/1497624/pappushyper.zip)

**Comment by bellezaodiada** [comment]
And this one is even worse:
Cinderella 2.9 build 1901:
![image](https://user-images.githubusercontent.com/16744124/33180581-7ad6bf18-d06d-11e7-9175-585546d3dd43.png)
Exported to CindyJS and viewed in Firefox 57.0 (64 bit)
![image](https://user-images.githubusercontent.com/16744124/33180640-a6bf8196-d06d-11e7-933e-db720cb6a8bd.png)

Cdy file is  
[pappuselipse.zip](https://github.com/CindyJS/CindyJS/files/1499714/pappuselipse.zip)

**Comment by elkins0** [comment]
I assert that the mistake lies within the Cinderella coding. Two foci and a point on the conic define both an ellipse and a hyperbola. The example shows the result of defining a hyperbola C0 first and then an ellipse C1 for the points A, B & C and also the result of defining an ellipse C2 first and then a hyperbola C3 for the points D, E & F. Have a look at the Construction Text window and notice that C0 & C1 both show Hyperbola(A,B,C) while C2 & C3 both show Ellipse(D,E,F). Then notice that the CindyJS window shows both the ellipse and hyperbola for points A, B & C but just a hyperbola for the points D, E & F. The Cinderella and CindyJS files are in the [pappusTestHE.zip](https://github.com/CindyJS/CindyJS/files/1500623/pappusTestHE.zip) file.
![pappustesthe cdy](https://user-images.githubusercontent.com/15895776/33195575-0c19c588-d096-11e7-8947-43fabad2e4be.png)
![pappustesthe-ct cdy](https://user-images.githubusercontent.com/15895776/33195576-1299cbce-d096-11e7-884d-514202871c68.png)
![pappustesthe html](https://user-images.githubusercontent.com/15895776/33195578-167ec6a4-d096-11e7-8e01-bd3be595ec21.png)

**Comment by elkins0** [comment]
Taking the `pos` and `color` attributes for C0, C1, C2 & C3 from the CindyJS `pappusTestHE.html` file from the above example and converting them into a _Mathematica_ statement to plot the conics shows that those values appear to be correct. So the mistake apparently lies somewhere in CindyJS. The Mathematica statement is shown here; its output follows:
```Mathematica
Show[Release[
  Hold[ContourPlot[
      x^2 xx + x xz + x xy y + y^2 yy + y yz + zz == 0, {x, -9.06, 
       18.14 + 1}, {y, -9.34, 4.1}, 
      AspectRatio -> (4.1 - (-9.34))/(18.14 + 1 - (-9.06)), 
      ContourStyle -> color, PlotPoints -> 40, 
      PerformanceGoal -> "Quality"]] /. {{name -> "C1", 
      xx -> -0.03449259105721611 + -5.10035293500915*^-18 I, 
      yy -> -0.08333333333333329 + -1.899092487899244*^-17 I, 
      zz -> 1.0, 
      xy -> -2.5320270526839145*^-18 + -2.358076689198932*^-18 I, 
      xz -> -0.13797036422886444 + -2.5117565118434464*^-17 I, 
      yz -> -0.33333333333333315 + -8.067985289436763*^-17 I, 
      color -> RGBColor[0.0, 1.0, 0.0]}, {name -> "C0", 
      xx -> 0.49999999999999994 + 1.9724043349150182*^-17 I, 
      yy -> -0.05173888658582399 + -8.779849427625047*^-18 I, 
      zz -> 0.6208666390298955 + -2.969858763927968*^-16 I, 
      xy -> 6.5383978800156555*^-34 + -1.6328000669688974*^-17 I, 
      xz -> 1.9999999999999998, 
      yz -> -0.20695554634329597 + -6.777539904987816*^-17 I, 
      color -> RGBColor[0.0, 0.0, 1.0]}, {name -> "C3", 
      xx -> 0.00782653212639395 + 6.382631234370851*^-19 I, 
      yy -> -0.0011547460061968327 + -6.579368616764591*^-18 I, 
      zz -> 1.0 + 2.465190328815662*^-32 I, xy -> 0.0, 
      xz -> -0.1800102389070608 + -2.4700784621051614*^-17 I, 
      yz -> -0.004618984024787331 + -2.6317474467058364*^-17 I, 
      color -> RGBColor[0.0, 1.0, 1.0]}, {name -> "C2", 
      xx -> 0.010612388947548618 + -9.132035475741092*^-19 I, 
      yy -> 0.028096750615927094 + 6.7768261512729*^-18 I, 
      zz -> 0.9999999999999999, xy -> -0.0, 
      xz -> -0.24408494579361814 + -3.6325280763290764*^-18 I, 
      yz -> 0.11238700246370838 + 2.71073046050916*^-17 I, 
      color -> RGBColor[1.0, 1.0, 0.0]}} /. v_Complex :> Re[v]], 
 Background -> RGBColor[{168, 176, 192}/255], ImageSize -> Full]
```
![pappustesthe](https://user-images.githubusercontent.com/15895776/33198736-0e86778a-d0aa-11e7-9f00-005e1f388265.png)

**Comment by strobelm** [comment]
Hi! The problem of all the examples is that ``PointOnConic`` is not implemented yet.

**Comment by bellezaodiada** [comment]
Are there plans to implement it before release 3.0? Is there a timescale?

**Comment by kortenkamp** [comment]
Best would be to just open an issue on GitHub - we will try to reach full compatibility.

**Comment by bellezaodiada** [comment]
Has this problem been fixed?
